jupyter_scheduler/scheduler.py

```diff
@@ -470,12 +530,41 @@ def list_job_definitions(self, query: ListJobDefinitionsQuery) -> ListJobDefinit
         return list_response
 
     def get_staging_paths(self, job_id: str) -> Dict[str, str]:
-        model = self.get_job(job_id)
+        model = None
+        with self.db_session() as session:
+            model = session.get(Job, job_id)
```

Comment:
This is actually a very relevant performance concern, and it's good that you caught this. When https://github.com/jupyter-server/jupyter-scheduler/pull/175 gets merged, building that model will require an expensive `file_id_manager.get_path()` call. In summary, the issue is that there are many scenarios in which we want to just fetch the job **record** instead of the `DescribeJob` model, because building that model involves additional compute load to bind attributes we don't need.

This is a cross-cutting concern outside of this class as well. We want to do the same in `OutputFilesManager.copy_from_staging()`.

Thus, I believe the solution here is that we add a new public method on scheduler called `get_job_record()`, which just returns the job record instead of the `DescribeJob` model. We should probably do the same for `get_job_definition_record()` just to preserve parity between job & job definition methods, and so we don't have to ask custom implementors to add another method twice.

Reply:
Thinking about this a bit more deeply, `OutputFilesManager.copy_from_staging()` shouldn't call `get_job_record()`, since the return type of this method would vary per implementation. Hence, we should just keep these as private methods and restrict their use to `Scheduler`.